ui/admin/app/routes/application.js

```diff
@@ -44,14 +44,21 @@ export default class ApplicationRoute extends Route {
       const fromName = transition?.from?.name;
       const toName = transition?.to?.name;
       const maybeModel = transition?.from?.attributes;
-      if (fromName !== toName && maybeModel?.hasDirtyAttributes) {
+      const dataModel = transition?.data?.model;
```

Comment:
Could you have done `const maybeModel = transition?.data?.model ?? transition?.from?.attributes;`?

That way you don't have to check two models in the if check or rolling back and you just decide which model to use in the very beginning

Reply:
great point, updated!